react-app/src/features/welcome/index.tsx

```diff
@@ -12,23 +12,33 @@ import { useFeatureFlag } from "@/hooks/useFeatureFlag";
 
 export const loader = (queryClient: QueryClient) => {
   return async () => {
+    // Check user query
+    const userFetch = await queryClient.fetchQuery({
+      queryKey: ["user"],
+      queryFn: () => getUser(),
+    });
+    // Verify location point for login redirect
+    const loginRedirect =
+      !["/login", "/faq"].includes(window.location.pathname) && !userFetch?.user;
+
+    if (loginRedirect) {
+      return <Navigate to="/login" />;
+    }
+
+    // Grab query string for possible errors
```

Comment:
1. I think the comments are unnecessary next to such clear code
2. `userFetch` can we renamed to `oneMacUser` or something clearer